### ISSUE TITLE
Ensure woocommerceanalytics_add_to_cart fires for stores that use blocks

### ISSUE THREAD

**Issue by haszari** [woocommerce/woocommerce-gutenberg-products-block#2030]
We'll use the `woocommerceanalytics` events (from [Jetpack](https://github.com/Automattic/jetpack/tree/master/modules/woocommerce-analytics)) for measuring conversion rate of the new block-based cart & checkout, and comparing with existing (shortcode) flow.

We need the `add_to_cart` event to fire correctly for stores using blocks – i.e. different ways of presenting products & `Add to cart` button, and any possible interaction with other blocks (e.g. cart/checkout, events fired async on next pageview etc).

See also #2031 #2032

**Comment by haszari** [comment]
This event seems to be getting dropped in Safari sometimes – it's consistent in Chrome or Firefox.

In Safari, I don't always see `add_to_cart` fire, for example:

- View a page with hand-picked or best sellers block. Also seeing issue with traditional (shortcode based) shop page.
- Click "add to cart" button for a product (in one of the above blocks).
   - Behind the scenes, this should add `add_to_cart` event data to the session, for sending on next page view.
- Refresh the page, or view another page.

On this page view I don't always see `add_to_cart` fire in Safari. In other browsers there's always 2 events: `add_to_cart` and `product_viewed`. In safari we get only one of these events (seems to be random which one fires).

**Comment by psealock** [comment]
>In Safari, I don't always see add_to_cart fire, for example:

In my testing, I'm getting `add_to_cart` every time, but not always `product_viewed`. This makes me think this condition is different in Safari, perhaps something to do with the way referrers are handled.

https://github.com/Automattic/jetpack/blob/8cb27fadb832c066a11b39051efed89bc4be407d/modules/woocommerce-analytics/classes/wp-woocommerce-analytics-universal.php#L331-L334

**Comment by haszari** [comment]
> In Safari, I don't always see add_to_cart fire, for example:

Update – this may be a false alarm. When filtering network requests tab in dev tools, it shows one request, but there are actually two. I'll continue testing ..

<img width="793" alt="Screen Shot 2020-03-30 at 4 00 30 PM" src="https://user-images.githubusercontent.com/4167300/77871134-a9a8cd00-729f-11ea-8adc-9c0c7d578567.png">

![Uploading Screen Shot 2020-03-30 at 4.00.20 PM.png…]()

**Comment by haszari** [comment]
I've tested this some more and am confident this event is firing for all our blocks. This is based on the PHP `woocommerce_add_to_cart` hook, which should be fired for all our block add to cart buttons via API.

I've tested a range of blocks on production build of WooCommerce Blocks + Jetpack Beta Bleeding edge (to get the latest Woo Analytics fixes), and I didn't see any issues. I tested these blocks on ephemeral site:

 - newest
 - on sale
 - products by attribute
 - top rated
 - products by tag

I also previously tested this on my dev environment:

- Tested purchase flow from traditional woo shop, category & product pages, with block cart/checkout.
- Tested add to cart from all products, featured product, hand-picked & best selling blocks.

Didn't see any issues, so happy to close this issue.

We'll need to retest this when we're ready to launch cart & checkout, and keep an eye on this for new releases of Woo and Jetpack – but we're good for now :)
